maintainer_guide.md

```diff
@@ -0,0 +1,75 @@
+## Torchvision maintainers guide
+
+This document aims at documenting user-facing policies / principles used when
+developing and maintaining torchvision. Other maintainer info (e.g. release
+process) can be found in the internal wiki.
+
+### What is public and what is private?
+
+For the Python API, torchvision largely follows the [torch core
+policy](https://github.com/pytorch/pytorch/wiki/Public-API-definition-and-documentation)
+which is consistent with other major packages (numpy, scikit-learn etc.).
+We recognize that his policy is somewhat imperfect for some edge cases, and that
+it's difficult to come up with an accurate technical definition. In broad terms,
+which are usually well understood by user, the policy is that:
+
+- modules that can be accessed without leading underscore are public
+- objects in a public file that don't have a leading underscore are public
+- objects that start with a leading underscore are private unless they're
+  exposed / aliased as public in a public `__init__.py` file
+- class attributes are public iff they have no leading underscore
+
+The public API has backward-compatible (BC) guarantees defined in our
+deprecation policy (see below). The private API has not BC guarantees.
+
+For C++, code is private. If a change breaks fbcode, fix fbcode or revert the
+change. We should be careful about models running in prod and relying on
+torchvision ops.
```

Comment:
Since this guide is not aimed purely at Meta employees, we should clarify this. Probably just preface this with "For Meta employees:"